src/components/dashboard/index.js

```diff
@@ -0,0 +1,329 @@
+/**
+ * @author johnworth
+ *
+ * The dashboard component.
+ *
+ * @module dashboard
+ */
+import React, { useState, useEffect } from "react";
+
+import { makeStyles } from "@material-ui/styles";
+import {
+    Button,
+    Card,
+    CardActions,
+    CardContent,
+    Divider,
+    Typography,
+} from "@material-ui/core";
+
+import { parse, format } from "date-fns";
+
+import { getMessage, withI18N, build as buildID } from "@cyverse-de/ui-lib";
+
+import messages from "./messages";
+import ids from "./ids";
+import * as constants from "./constants";
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        flexGrow: 1,
+    },
+    dividerRoot: {
+        marginBottom: 15,
+    },
+    footer: {
+        width: "100%",
+        height: 128, // This is needed to get the vertical scrolling to stop cutting off the bottom of the content.
+
+        [theme.breakpoints.down("sm")]: {
+            height: 32,
+        },
+    },
+    section: {
+        marginTop: 25,
+        marginBottom: 50,
+    },
+    sectionItems: {
+        display: "flex",
+        flexWrap: "wrap",
+        justifyContent: "flex-start",
+        padding: 10,
+
+        // Try to eek as much space out of the iPhone SE cards as possible.
+        [theme.breakpoints.down("sm")]: {
+            padding: 0,
+            justifyContent: "center",
+        },
+    },
+    subtitle: {
+        marginBottom: 15,
+    },
+    gridRoot: {
+        overflow: "auto", // Needed for vertical scrolling.
+        height: "100vh", // Needed to get the vertical scrolling working.
+        padding: "0 30px 0 30px",
+
+        [theme.breakpoints.down("sm")]: {
+            padding: "0 5px 0 5px",
+        },
+    },
+    dashboardCard: {
+        width: 450,
+        height: 225,
+        display: "flex",
+        flexDirection: "column",
+        margin: 5,
+
+        [theme.breakpoints.down("sm")]: {
+            width: 300,
+        },
+    },
+    actionsRoot: {
+        marginLeft: "auto",
+    },
+}));
+
+const getOrigination = (kind, content) => {
+    let origination;
+    let date;
+
+    switch (kind) {
+        case constants.KIND_ANALYSES:
+            origination = getMessage("startedBy");
+            date = content.start_date;
+            break;
+        case constants.KIND_APPS:
+            if (content.integration_date) {
+                origination = getMessage("integratedBy");
+                date = content.integration_date;
+            } else {
+                origination = getMessage("editedBy");
+                date = content.edited_date;
+            }
+            break;
+        case constants.KIND_FEEDS:
+            origination = getMessage("publishedBy");
+            date = content.date_added;
+            break;
+        default:
+            origination = getMessage("by");
+            date = new Date();
+    }
+
+    date = parse(date);
+    date = format(date, "MMMM Do YYYY, h:mm:ss a");
```

Comment:
This can done using the `formatDate` in `ui-lib`. But that can wait until https://github.com/cyverse-de/ui-lib/pull/33 is merged.